clang/runtime/dpct-rt/include/dpl_extras/algorithm.h

```diff
@@ -1449,8 +1449,664 @@ inline void segmented_sort_pairs_by_two_pair_sorts(
+
+          __reduce_out_histograms<BinType>(hacc_private,
+                                           __wgroup_idx * num_bins, bins,
+                                           num_bins, __self_item);
+        });
+  });
+  e.wait();
+}
+
+template <typename Policy, typename Iter1, typename Iter2, typename Size,
+          typename IdxHashFunc, typename... Range>
+inline Iter2
+__histogram_general_select_best(Policy &&policy, Iter1 first, Iter1 last,
+                                Iter2 histogram_first, const Size &num_bins,
+                                IdxHashFunc func, Range &&...opt_range) {
+  using __histo_value_type = typename ::std::iterator_traits<Iter2>::value_type;
+  auto __local_mem_size =
+      policy.queue()
+          .get_device()
+          .template get_info<sycl::info::device::local_mem_size>();
+  constexpr ::std::uint8_t __max_registers = 16;
+
+  // Temporary use of stable non-public API from oneDPL,  this function will be
+  // replaced with oneDPL call in an upcoming release.
+  auto keep_bins = oneapi::dpl::__ranges::__get_sycl_range<
+      oneapi::dpl::__par_backend_hetero::access_mode::write, Iter2>();
+  auto bins_buf = keep_bins(histogram_first, histogram_first + num_bins);
+
+  oneapi::dpl::fill(std::forward<Policy>(policy), bins_buf.all_view().begin(),
+                    bins_buf.all_view().end(), __histo_value_type(0));
+  auto N = last - first;
+  if (N > 0)
+  {
+    // Temporary use of stable non-public API from oneDPL,  this function will be
+    // replaced with oneDPL call in an upcoming release.
+    auto keep_input = oneapi::dpl::__ranges::__get_sycl_range<
+        oneapi::dpl::__par_backend_hetero::access_mode::read, Iter1>();
+    auto input_buf = keep_input(first, last);
+
+
+    if (num_bins < __max_registers) {
+      //std::cout<<"registers"<<std::endl;
+
+      // If bins fit into registers, use register private accumulation
+      __histogram_general_registers_local_reduction<1024, 32, 16,
+                                                    __histo_value_type>(
+          ::std::forward<Policy>(policy), input_buf.all_view(),
+          bins_buf.all_view(), num_bins, func,
+          std::forward<Range...>(opt_range)...);
+    } else if (num_bins * sizeof(__histo_value_type) < __local_mem_size) {
+      // If bins fit into SLM, use local atomics
+      //std::cout<<"local atomics"<<std::endl;
+
+      // Experimentally determined iters per work-item
+      if (N <= 524288) {
+        __histogram_general_local_atomics<1024, 4, __histo_value_type>(
+            ::std::forward<Policy>(policy), input_buf.all_view(),
+            bins_buf.all_view(), num_bins, func,
+            std::forward<Range...>(opt_range)...);
+      } else {
+        __histogram_general_local_atomics<1024, 32, __histo_value_type>(
+            ::std::forward<Policy>(policy), input_buf.all_view(),
+            bins_buf.all_view(), num_bins, func,
+            std::forward<Range...>(opt_range)...);
```

Comment:
`std::` -> `::std::` here and in the surrounding conditionals.

Reply:
fixed.